### PR TITLE
refactored NetworkRecipient interface

### PR DESCRIPTION
**Pull request by pompon0**
Refactored NetworkRecipient interface so that it supports multiple message types and easy mocking. Currently we have an enum for all-messages and an enum for all-responses, which requires an unwrap of right response variant at every call site.

### CODE REVIEW

File: chain/client/src/test_utils.rs [hunk 1]
```diff
@@ -592,13 +606,11 @@ pub fn setup_mock_all_validators(
         let client_addr = ClientActor::create(|ctx| {
             let client_addr = ctx.address();
             let _account_id = account_id.clone();
-            let pm = PeerManagerMock::mock(Box::new(move |msg, _ctx| {
+            let pm = PeerManagerMock{handle:Box::new(move |msg, _ctx| {
```
Comment (matklad) [inline chain/client/src/test_utils.rs:609]: Missing whitespace. I think that's the second time we manually construct a PereManagerMock, so let's maybe add `PeerManagerMock::new` which accepts an `impl Fn` and does Boxing and type-erasure internally.

**Review by matklad — APPROVED** [review]
LGTM, but lets add PeerManagerMock::new

File: chain/client/src/test_utils.rs (hunk 1 above)
Comment (pompon0) [reply chain/client/src/test_utils.rs]: done

Comment (matklad) [reply chain/client/src/test_utils.rs]: Also, that's a general pattern: you provide `impl Trait` interface at the API level, and erase to `Box<dyn Trait>` internally -- usually gives a nicer to use API, and shields the user of the details of boxing

**Pull request merged**